### ISSUE TITLE
RegExp in Domain Pattern

### ISSUE THREAD

**Issue by grenzor** [stoically/temporary-containers#120]
Currently, wildcard is supported in domain patterns but not Regexp. It can be useful for the ability to use regular expressions in domain patterns for users who want more specificity and advanced ways of choosing what domains or site paths a "per-site" container settings applies to (or not apply to). Just a basic example: `^((?!yahoo.com|github.com).)*$` as domain pattern for a specific per-site setting will apply those settings and match them for all sites except yahoo and github. 

Could also be a way for advanced users to craft and use expressions in a way similar to what you want to achieve in #116.

**Comment by stoically** [comment]
Yep, good idea - was thinking the same. The way I currently think about implementing it is that, if the Domain Pattern starts and ends with `/` it is treated as RegExp and matches the full URL instead of just the Domain; which will be documented in the wiki for Advanced users.

**Comment by stoically** [comment]
Though; that doesn't help with the use-case I had for #116 in mind yet:
- Same tmp as long as path `/[^/]+/[^/]+(\/.+)?` is matched
- *But* new tmp if $1 or $2 changes

Maybe some sort of "modifier"? Hmm.

**Comment by grenzor** [comment]
>Domain Pattern starts and ends with / it is treated as RegExp and matches the full URL instead of just the Domain

Sounds good to me. Regexp can be pretty flexible so it should be able to cover a lot of use cases. Supporting regex flags like `/i` for case insensitivity can also be useful.

**Comment by stoically** [comment]
> Supporting regex flags like /i for case insensitivity can also be useful.

Ah, true, for that case it might make sense to treat the pattern as RegExp as soon as it starts with `/` - which is OK since it doesn't interfere with normal Domain Pattern matching anyway. However, I think incoming requests are always lower-cased from Firefox anyway, might be something worth investigating and documenting.

**Comment by stoically** [comment]
But of course Paths (and Queries) in URLs aren't lower-cased, hehe.
